### ISSUE TITLE
regression: global namespace no longer matters

### ISSUE THREAD

**Issue by starpit** [openwhisk/openwhisk-client-js#39]
at some point a change was made that breaks code that relies on the global `namespace` field, i.e. the one provided with the openwhisk constructor.

now, the namespace is either the per-request `options.namespace` or `names.default_namespace()`, the latter of which does not pay attention to the global option:

```
const default_namespace = () => {
  return process.env['__OW_NAMESPACE'] || '_'
}
```

this breaks our code. luckily we are still pre-production, or this would've been a nightmare.

perhaps this: 9269e96534d5286c69d8dcc9d97043035ad4c897

**Comment by starpit** [comment]
btw, this was very difficult to debug, because the error reporting from the npm is scant. perhaps have an option to report verbose errors, where the details of the (failed) request are emitted?

**Comment by rabbah** [comment]
That would help. In particular don't swallow the error response from the controller if that happening - the container code would help debug user problems.

**Comment by jthomas** [comment]
I've pushed a fix to remove the regression and published a new version. I've added a unit test to ensure this doesn't happen again. 

The precedence order for calculating namespaces is: options.namespace, global_options.namespace, env param, ‘_'.

I agree that we should have a debug property to dump the raw requests & responses. I did open a ticket a while ago to add this.
https://github.com/openwhisk/openwhisk-client-js/issues/15
